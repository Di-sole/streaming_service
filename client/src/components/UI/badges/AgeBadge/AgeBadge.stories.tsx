import { Meta, StoryObj } from '@storybook/react';
import './AgeBadge.module.scss';
import Badge from './AgeBadge';

const meta: Meta<typeof Badge> = {
    title: 'Badges/AgeBadge',
    component: Badge,
};

export default meta;

type Story = StoryObj<typeof Badge>;

export const AgeBadge: Story = {
    args: {
        value: '8',
    },
};
