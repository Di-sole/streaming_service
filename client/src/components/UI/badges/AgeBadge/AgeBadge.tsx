import styles from './AgeBadge.module.scss';

interface AgeBageProps {
    value: string;
    className?: string;
}

const AgeBadge = ({ value, className }: AgeBageProps) => {
    return (
        <div className={[styles.badge, className].join(' ')} data-testid={'ageBadge'}>
            {value}+
        </div>
    );
};

export default AgeBadge;
