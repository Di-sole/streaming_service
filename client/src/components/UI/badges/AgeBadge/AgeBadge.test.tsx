import { render, screen } from '@testing-library/react';
import AgeBadge from './AgeBadge';

describe('AGEBADGE SNAPSHOTS TESTS', () => {
    test('AgeBadge should not be changed', () => {
        render(<AgeBadge value="8" />);

        const id = screen.getByTestId('ageBadge');
        expect(id).toMatchSnapshot();
    });
});
