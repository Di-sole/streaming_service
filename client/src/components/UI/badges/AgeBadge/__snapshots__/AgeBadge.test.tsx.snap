// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AGEBADGE SNAPSHOTS TESTS AgeBadge should not be changed 1`] = `
<div
  class="badge "
  data-testid="ageBadge"
>
  8
  +
</div>
`;
